src/sentry/api/serializers/models/event.py

```diff
@@ -31,41 +35,61 @@ def _get_entries(self, event, user, is_public=False):
                 'data': data,
                 'type': interface.get_alias(),
             }
-            interface_list.append((interface, entry))
+
+            api_meta = None
+            if meta.get(key):
+                api_meta = interface.get_api_meta(meta[key], is_public=is_public)
+
+            interface_list.append((interface, entry, api_meta))
+
         interface_list.sort(
             key=lambda x: x[0].get_display_score(), reverse=True)
 
-        return [i[1] for i in interface_list]
+        return (
+            [i[1] for i in interface_list],
+            {k: {'data': i[2]} for k, i in enumerate(interface_list)}
+        )
+
+    def _get_interface_with_meta(self, event, name, is_public=False):
+        interface = event.interfaces.get(name)
+        if not interface:
+            return (None, None)
+
+        data = interface.get_api_context(is_public=is_public)
+        event_meta = event.data.get('_meta') or {}
+        if not data or not event_meta.get(name):
+            return (data, None)
+
+        api_meta = interface.get_api_meta(event_meta[name], is_public=is_public)
+        # data might not be returned for e.g. a public HTTP repr
+        if not api_meta:
+            return (data, None)
+
+        return (data, meta_with_chunks(data, api_meta))
 
     def get_attrs(self, item_list, user, is_public=False):
         Event.objects.bind_nodes(item_list, 'data')
 
         results = {}
         for item in item_list:
-            user_interface = item.interfaces.get('sentry.interfaces.User')
             # TODO(dcramer): convert to get_api_context
-            if user_interface:
-                user_data = user_interface.to_json()
```

Comment:
**Important**: This was `.to_json()` and is now `.get_api_context()`. The UI expects `user.ip_address` and API context was dead code. The only use of `ipAddress` is in _groupTagValues.jsx_.